### PR TITLE
Add initial implentation

### LINKED ISSUE #1: Move elgamal encryption from `phoenix-core`
#### Summary
Move the elgamal encryption scheme from `phoenix-core` into its own library so that it can be used by other crates as well.

#### Relevant Context
This is needed for zedger.
See [phoenix#255](https://github.com/dusk-network/phoenix/issues/255).

### PR DESCRIPTION
**Pull request by moCello**
Resolves #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+To decrypt the ciphertext $(c_1, c_2)$ Bob will use his secret-key $sk_B$:
+
+
+- Compute $c_2 - c_1 * sk_B = m$
+
+This is true because:
+$$
+c_2 - c_1 * sk_B = m + PK_B * r - (r * G * sk_B) = m + PK_B * r - PK_B * r = m
+$$
```
Comment (Neotamandua) [inline README.md:56]: This is not rendering properly on the md (at least for me)

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    let ciphertext_2 = plaintext + public_key * r;
+
+    (ciphertext_1, ciphertext_2)
+}
+
+/// Uses the given `secret_key` to decrypt the given `ciphertext` to the
+/// original plaintext.
+///
+/// ## Return
+/// Returns the `[JubJubExtended]` plaintext.
```
Comment (moCello) [inline src/lib.rs:36]: Will fix this:
```suggestion
/// Returns the [`JubJubExtended`] plaintext.
```

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        let ciphertext_2 = composer.component_add_point(plaintext, r_point);
+
+        Ok((ciphertext_1, ciphertext_2))
+    }
+
+    /// Uses the given `secret_key` to decrypt the given `ciphertext` to the
+    /// original plaintext in a gadget that can be used in a plonk-circuit.
+    ///
+    /// ## Return
+    /// Returns the `[WitnessPoint]` plaintext.
```
Comment (moCello) [inline src/lib.rs:82]: Will fix this
```suggestion
    /// Returns the [`WitnessPoint`] plaintext.
```

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+#![deny(rustdoc::broken_intra_doc_links)]
+#![deny(clippy::pedantic)]
+
+use dusk_jubjub::{JubJubExtended, JubJubScalar, GENERATOR};
+
+/// Uses the given `public_key` and a fresh random number `r` to encrypt a
+/// plaintext [`JubJubExtended`].
+///
+/// ## Return
+/// Returns the ciphertext tuple `([JubJubExtended], [JubJubExtended])`.
```
Comment (moCello) [inline src/lib.rs:19]: Will fix this
```suggestion
/// Returns the ciphertext tuple `(JubJubExtended, JubJubExtended)`.
```

**Review by Neotamandua — APPROVED** [review]
LGTM

**Pull request merged**